x-pack/plugins/apm/public/components/shared/transactions_table/index.tsx

```diff
@@ -129,9 +131,23 @@ export function TransactionsTable({
 
   const { transactionType, serviceName } = useApmServiceContext();
 
+  const dataSource = usePreferredDataSourceAndBucketSize({
```

Comment:
nit: you've called this `preferred` almost everywhere else (one occurrence of `dataSourceOptions` which also invokes my OCD-lite)

```suggestion
  const preferred = usePreferredDataSourceAndBucketSize({
```

Reply:
agreed (although `preferred` is not ideal either).